tidy3d/web/cli/converter.py

```diff
@@ -0,0 +1,1332 @@
+            x = Lines[i][8:].split(";")[0][:-1]
+        elif Lines[i][5:7] == "y'":
+            y = Lines[i][8:].split(";")[0][:-1]
+        elif Lines[i][5:7] == "z'":
+            z = Lines[i][8:].split(";")[0][:-1]
+        elif Lines[i][5:10] == "x_min":
+            x_min = Lines[i][12:].split(";")[0][:-1]
+            xl = x_max + "-" + x_min
+        elif Lines[i][5:10] == "x_max":
+            x_max = Lines[i][12:].split(";")[0][:-1]
+            xl = x_max + "-" + x_min
+        elif Lines[i][5:10] == "y_min":
+            y_min = Lines[i][12:].split(";")[0][:-1]
+            yl = y_max + "-" + y_min
+        elif Lines[i][5:10] == "y_max":
+            y_max = Lines[i][12:].split(";")[0][:-1]
+            yl = y_max + "-" + y_min
+        elif Lines[i][5:10] == "z_min":
+            z_min = Lines[i][12:].split(";")[0][:-1]
+            zl = z_max + "-" + z_min
+        elif Lines[i][5:10] == "z_max":
+            z_max = Lines[i][12:].split(";")[0][:-1]
+            zl = z_max + "-" + z_min
+        elif Lines[i][5:11] == "x span":
+            xl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:11] == "y span":
+            yl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:11] == "z span":
+            zl = Lines[i][13:].split(";")[0][:-1]
+        elif Lines[i][5:19] == "mode selection":
+            mode = Lines[i][22:].split(";")[0][:-1]
+            if mode == "'fundamental TM mode'":
+                mode = "1"
+                num_modes = "2"
+            else:
+                mode = "0"
+        elif Lines[i][5:22] == "wavelength center":
+            wave_center = Lines[i][24:].split(";")[0][:-1] + "*1e6"
+        elif Lines[i][5:20] == "wavelength span":
+            wave_span = Lines[i][22:].split(";")[0][:-1] + "*1e6"
+        elif Lines[i][5:21] == "frequency points":
+            num_freqs = Lines[i][23:].split(";")[0][:-1]
+            freqs = (
+                "np.linspace("
+                + wave_center
+                + "-"
+                + wave_span
+                + "/2,"
+                + wave_center
+                + "+"
+                + wave_span
+                + "/2,"
+                + num_freqs
+                + ")"
+            )
+        i += 1
+        if i == len(Lines):
+            break
+    mode_monitor_string += num_modes + ") # NOTE: check if solving for correct mode!\n"
+    mode_monitor_string += name.replace(" ", "_") + " = td.ModeMonitor(\n"
+    x, y, z, xl, yl, zl = _to_um([x, y, z, xl, yl, zl])
+    mode_monitor_string += "\tcenter=(" + x + "," + y + "," + z + "),\n"
+    mode_monitor_string += "\tsize=(" + xl + "," + yl + "," + zl + "),\n"
+    mode_monitor_string += "\tfreqs=" + freqs + ",\n"
+    mode_monitor_string += "\tmode_spec=" + mode_spec_name + ",\n"
+    mode_monitor_string += "\tname='" + name + "',\n"
+    mode_monitor_string += ")\n"
+    return mode_monitor_string, name.replace(" ", "_")
+
+
+def lsf_reader(filename=str) -> None:
```

Comment:
to annotate the `filename` as a string, one does `filename: str`. what is here currently assigns `filename` to the `str` type, if it is not provided so that would have some weird unintended consequences if called without an argument.